app/app_xdg.go

```diff
@@ -20,8 +20,15 @@ import (
 	"fyne.io/fyne/v2/theme"
 )
 
+var currentVariant atomic.Value
+
 func defaultVariant() fyne.ThemeVariant {
-	return findFreedestktopColorScheme()
+	val := currentVariant.Load()
+	if val == nil {
+		return theme.VariantDark
+	}
+
+	return val.(fyne.ThemeVariant)
```

Comment:
With the suggestion above, this would just be something like this (default/zero value is zero, aka `theme.VariantDark`):
```suggestion
	return fyne.ThemeVariant(currentVariant.Load())
```